spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/SpecTestFixture.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright (c) 2009 - ${YEAR} Red Hat, Inc.
+ *
+ *  This software is licensed to you under the GNU General Public License,
+ *  version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ *  implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ *  FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ *  along with this software; if not, see
+ *  http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ *  Red Hat trademarks are not licensed under GPLv2. No permission is
+ *  granted to use or replicate Red Hat trademarks that are incorporated
+ *  in this software or its documentation.
+ */
+
+package org.candlepin.spec.bootstrap.client;
+
+import org.candlepin.ApiException;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+import java.util.*;
+
+
+public abstract class SpecTestFixture {
+    public final static String OWNER = "Owner";
+
+    public Map<String, SpecTestClient> clientMap ;
+    protected ApiClientFactory apiClientFactory;
+
+    @BeforeAll
+    void setUp() {
+        clientMap = new HashMap<>();
+        apiClientFactory = new ApiClientFactory();
+    }
+
+    @AfterAll
+    void tearDown() throws ApiException {
```

Comment:
I'm happy to go with whichever y'all want to.
@nikosmoum  and @Ceiu ?

Reply:
If we have an intention to be able to run these against any arbitrary server and minimize the amount of data left over from that, then we'll need some kind of cleanup on the order of what the current spec tests do. If that is no longer a concern or design constraint, then we can drop it.

If it's not something that's been discussed, I would lean toward doing cleanup as that's what we already have and it's, at least theoretically, not that hard to deal with.

Reply:
We have gone a bit back and forth when discussing this, but I think we wanted to avoid the overhead from doing granular cleanup, so:
- We make sure we have good randomization utility functions (similar to the `random_string()` in rspec) for creating data with unique attributes and avoid duplication errors.
- When running the tests sequentially, we would be running a SQL TRUNCATE on all tables (with exceptions for tables that hold data that drive business logic, such as cp_consumer_type) between each test.
- When running the tests in parallel, we would not be doing any cleanup.

But all that is for later tasks. For now, we should probably start without any cleanup.

Reply:
If we're going to be doing TRUNCATES, we literally cannot run the spec suite against arbitrary deployments, as we'd be destroying any pre-existing data that might be there with it. 

If that's understood and accepted, fine, however it calls into question the value of isolating this to its own project/package completely separate from the main code base. I get that it helps us avoid accidentally violating spec test principles, but ... not much else.

Reply:
Running the spec tests against arbitrary deployments wasn't something we'd discussed, but it could be useful. I think the most important thing here is to avoid performance overhead from having to track and clean test data. So we could very well just not do any cleanup.

Reply:
I took out the cleanup for now.